akka-actor-tests/src/test/scala/akka/io/UdpConnectedIntegrationSpec.scala

```diff
@@ -6,15 +6,24 @@ package akka.io
 
 import java.net.InetSocketAddress
 
-import akka.testkit.{ AkkaSpec, ImplicitSender, TestProbe }
-import akka.util.ByteString
 import akka.actor.ActorRef
 import akka.testkit.SocketUtil.temporaryServerAddresses
 import akka.testkit.WithLogCapturing
+import akka.testkit.AkkaSpec
+import akka.testkit.ImplicitSender
+import akka.testkit.TestProbe
+import akka.util.ByteString
+
 import scala.concurrent.duration._
 
 class UdpConnectedIntegrationSpec extends AkkaSpec("""
     akka.loglevel = DEBUG
+    akka.actor.debug.lifecycle = on
+    akka.actor.debug.autoreceive = on
+    akka.io.udp-connected.trace-logging = on
+    # issues with dns resolution of non existent host hanging with the
+    # Java native host resolution
+    akka.io.dns.resolver = async-dns
```

Comment:
This PR now does two things: 
1. changes this to async-dns because of JDK blocking resolution hanging
2. fixes issues in udp and tcp connection for adapted old DNS protocol when using async-dns

Reply:
should we explicitly recommend against this in docs as well then?